### DISCUSSION TITLE
How do I run a minecraft server through a bash file instead of a java command? (Ubuntu)

### DISCUSSION THREAD

**Discussion by Sean0203** [pufferpanel/pufferpanel#1223] [Q&A]
I have a minecraft modpack (medieval mc fabric edition) that needs to be launched with a start.sh file. I've tried changing the run command to "bash start.sh", but the console was empty and nothing seemed to happen in the background (checked with top). However, if I run the command directly on the terminal the server starts up normally. I hope someone can help me on this.

[edit] Just in case, here is the link to the start.sh file: https://file.io/szRwvrisHsbA

**Comment by ozhound** [comment]
Quite often, those server start scripts are an all-in-one way to install
and run the server.

with these packs, i initially run the startup script on my pc and then copy
the directory over to puffer.

as for your startup line, and i run that exact server (although via MC
Server Soft on windows) the minecraft fabric template should work.

i suggest edit the server definition to change the run configuration to
something like this

${java} -Xmx${memory}M -XX:+UseG1GC -XX:+ParallelRefProcEnabled
> -XX:MaxGCPauseMillis=200 -XX:+UnlockExperimentalVMOptions
> -XX:+DisableExplicitGC -XX:+AlwaysPreTouch -XX:G1NewSizePercent=30
> -XX:G1MaxNewSizePercent=40 -XX:G1HeapRegionSize=32M -XX:G1ReservePercent=20
> -XX:G1HeapWastePercent=5 -XX:G1MixedGCCountTarget=4
> -XX:InitiatingHeapOccupancyPercent=15 -XX:G1MixedGCLiveThresholdPercent=90
> -XX:G1RSetUpdatingPauseTimePercent=5 -XX:SurvivorRatio=32
> -XX:+PerfDisableSharedMem -XX:MaxTenuringThreshold=1 -Dusing.aikars.flags=
> https://mcflags.emc.gs -Daikars.new.flags=true  -jar
> fabric-server-launcher.jar
>
